### DISCUSSION TITLE
Save color .plу file

### DISCUSSION THREAD

**Discussion by Emorh** [pyvista/pyvista#4745] [Q&A] [answered]
Hi everyone,

I don't quite understand how I can save a colored object. I have a color object "obj.ply" on my disk and can load it with the following code:

```python
reader = pv.get_reader("obj.ply")
mesh = reader.read()
```

if I plot this object, then all the colors in it are saved

```python
mesh.plot()
```

but if I save it all color information is lost
```python
mesh.save("save.ply")
```

**Comment by tkoyama010** [comment, marked as answer]
@Emorh 
We can save color using the argument `texture` (see [pyvista.PolyData.save](https://pyvista.github.io/pyvista-docs-dev-ja/api/core/_autosummary/pyvista.PolyData.save.html#pyvista.PolyData.save)).
Could you try the following code?

```python
>>> import pyvista as pv
>>> reader = pv.get_reader("obj.ply")
>>> mesh = reader.read()
>>> print(mesh.array_names)
['RGB']
>>> # Write a single texture array to file when using a PLY file.
>>> mesh.save("save.ply", texture=mesh.array_names[0])
```
